lib/router/index.js

```diff
@@ -203,6 +204,12 @@ proto.handle = function handle(req, res, out) {
       return;
     }
 
+    // max sync stack
+    if (++sync > 100) {
```

Comment:
Without looking at the current v8 stack frames, there is no way to know if you still have 100 frames left to use. Also params will add/remove to how many frames are used in each iteration, I wouldn't think. How did you come up with the number 100 for this?

Reply:
https://github.com/pillarjs/router/commit/e96ebf5ab30661905107e1ff15250baeb4a47b9f